fastlane_bot/events/event_gatherer.py

```diff
@@ -48,16 +51,47 @@ def get_all_events(self, from_block: int, to_block: int):
                 from_block_ = 0
             else:
                 from_block_ = from_block
-            coroutines.append(self._get_events_for_topic(from_block_, to_block, sub))
+            coroutines.append(self._get_events_for_subscription(from_block_, to_block, sub))
         results = asyncio.get_event_loop().run_until_complete(asyncio.gather(*coroutines))
         return list(chain.from_iterable(results))
 
-    async def _get_events_for_topic(self, from_block: int, to_block: int, subscription: Subscription):
-        events = await self._w3.eth.get_logs(filter_params={
-            "fromBlock": from_block,
-            "toBlock": to_block,
-            "topics": [subscription.topic]
-        })
-        return [subscription.parse_log(event) for event in events]
+    async def _get_events_for_subscription(self, from_block: int, to_block: int, subscription: Subscription):
+        return [subscription.parse_log(log) for log in await self._get_logs_for_topics(from_block, to_block, [subscription.topic])]
 
+    async def _get_logs_for_topics(self, from_block: int, to_block: int, topics: List[str]):
+        chunk_size = BLOCK_CHUNK_SIZE_MAP[self._blockchain]
+        if chunk_size > 0:
+            return await self._get_logs_iterative(from_block, to_block, topics, chunk_size)
+        else:
+            return await self._get_logs_recursive(from_block, to_block, topics)
 
+    async def _get_logs_iterative(self, from_block: int, to_block: int, topics: List[str], chunk_size: int) -> list:
+        block_numbers = list(range(from_block, to_block + 1, chunk_size)) + [to_block + 1]
+        log_lists = await asyncio.gather([
+            self._w3.eth.get_logs(filter_params={
+                "fromBlock": r[0],
+                "toBlock": r[1],
+                "topics": topics
+            })
+            for r in zip(block_numbers, map(lambda n: n - 1, block_numbers[1:]))
+        ])
+        return [log for log_list in log_lists for log in log_list]
+
+    async def _get_logs_recursive(self, from_block: int, to_block: int, topics: List[str]) -> list:
+        if from_block <= to_block:
+            try:
+                return await self._w3.eth.get_logs(filter_params={
+                    "fromBlock": from_block,
+                    "toBlock": to_block,
+                    "topics": topics
+                })
+            except Exception as e:
+                assert "eth_getLogs" in str(e), str(e)
+                if from_block < to_block:
+                    mid_block = (from_block + to_block) // 2
+                    log_lists = await asyncio.gather(
+                        self._get_logs_recursive(from_block, mid_block, topics),
+                        self._get_logs_recursive(mid_block + 1, to_block, topics)
+                    )
+                    return [log for log_list in log_lists for log in log_list]
+        return []
```

Comment:
We probably need to raise an error here, because we otherwise may return with missing results but nothing to indicate that.

Reply:
Let's print exception instead of raising it, so that the process is not interrupted. Done so